src/utils.js

```diff
@@ -76,29 +97,34 @@ export const indexOf = (array, value) => {
  */
 export const createReplacer = (replacer, circularReplacer) => {
   let cache = [];
+  let initd = 0;
 
+  const callable = typeof replacer === 'function';
   const getCircular = circularReplacer || getCircularValue;
 
   return function(key, value) {
-    if (cache.length) {
-      const locationOfThis = indexOf(cache, this);
+    let replacedValue = value;
+    if (initd) { // instead of read property of object every time
+      const locationOfThis = lastIndexOf(cache, this);
 
       if (~locationOfThis) {
-        cache = first(cache, locationOfThis + 1);
+        cache.length = locationOfThis + 1; // waste for create new array, we drop items at tail
```

Comment:
in my perf testing, this was actually shown to be signficantly slower. do you have alternative numbers to back up this switch?